### PR TITLE
WINDUP-667: FileContent.from()

### PR DESCRIPTION
**Pull request by mbriskar**
Changed FileContent so it is possible to use it along with JavaClass or XmlFile. fileNamePatter is not mandatory anymore

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/232/<h2>Build result: FAILURE</span></h2>[...truncated 23452 lines...][JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/victi.ms/pom.xml to org.jboss.windup.quickstarts/windup-victims/2.3.0-SNAPSHOT/windup-victims-2.3.0-SNAPSHOT.pomchannel stoppedArchiving artifactsIRC notifier plugin: Sending notification to: #windupSetting status of ee51a262fb43b31c22463fafe6c27d515e8484e6 to FAILURE with url http://10.3.8.5/job/windup-core-PR-dependents-check/232/ and message: Merged build finished.FileNotFoundException means that the credentials Jenkins is using is probably wrong. Or that something is really wrong with github.java.io.FileNotFoundException: https://api.github.com/repos/windup/windup/statuses/ee51a262fb43b31c22463fafe6c27d515e8484e6	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1625)	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)	at org.kohsuke.github.Requester.parse(Requester.java:383)	at org.kohsuke.github.Requester._to(Requester.java:185)	at org.kohsuke.github.Requester.to(Requester.java:160)	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:702)	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:714)	at org.jenkinsci.plugins.ghprb.GhprbRepository.createCommitStatus(GhprbRepository.java:140)	at org.jenkinsci.plugins.ghprb.GhprbRepository.createCommitStatus(GhprbRepository.java:122)	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onCompleted(GhprbBuilds.java:118)	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:27)	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:12)	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:199)	at hudson.model.Run.execute(Run.java:1802)	at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:531)	at hudson.model.ResourceController.execute(ResourceController.java:89)	at hudson.model.Executor.run(Executor.java:240)
Test FAILed.

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/864/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-base-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/864/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-base-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/864/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport/org.jboss.windup.rules.apps.condition/FileContentTest/testFileContentScan/'><strong>org.jboss.windup.rules.apps.condition.FileContentTest.testFileContentScan</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/233/<h2>Build result: FAILURE</span></h2>[...truncated 23733 lines...][JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/victi.ms/pom.xml to org.jboss.windup.quickstarts/windup-victims/2.3.0-SNAPSHOT/windup-victims-2.3.0-SNAPSHOT.pomchannel stoppedArchiving artifactsIRC notifier plugin: Sending notification to: #windupSetting status of ee51a262fb43b31c22463fafe6c27d515e8484e6 to FAILURE with url http://10.3.8.5/job/windup-core-PR-dependents-check/233/ and message: Merged build finished.FileNotFoundException means that the credentials Jenkins is using is probably wrong. Or that something is really wrong with github.java.io.FileNotFoundException: https://api.github.com/repos/windup/windup/statuses/ee51a262fb43b31c22463fafe6c27d515e8484e6	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1625)	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)	at org.kohsuke.github.Requester.parse(Requester.java:383)	at org.kohsuke.github.Requester._to(Requester.java:185)	at org.kohsuke.github.Requester.to(Requester.java:160)	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:702)	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:714)	at org.jenkinsci.plugins.ghprb.GhprbRepository.createCommitStatus(GhprbRepository.java:140)	at org.jenkinsci.plugins.ghprb.GhprbRepository.createCommitStatus(GhprbRepository.java:122)	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onCompleted(GhprbBuilds.java:118)	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:27)	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:12)	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:199)	at hudson.model.Run.execute(Run.java:1802)	at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:531)	at hudson.model.ResourceController.execute(ResourceController.java:89)	at hudson.model.Executor.run(Executor.java:240)
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/865/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/234/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/866/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-base-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/866/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-base-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/866/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport/org.jboss.windup.rules.apps.condition/FileContentTest/testFileContentScan/'><strong>org.jboss.windup.rules.apps.condition.FileContentTest.testFileContentScan</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/235/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/867/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-base-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/867/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-base-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/867/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport/org.jboss.windup.rules.apps.condition/FileContentTest/testFileContentScan/'><strong>org.jboss.windup.rules.apps.condition.FileContentTest.testFileContentScan</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/237/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/869/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-base-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/869/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-base-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/869/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport/org.jboss.windup.rules.apps.condition/FileContentTest/testFileContentScan/'><strong>org.jboss.windup.rules.apps.condition.FileContentTest.testFileContentScan</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/238/
Test PASSed.

**Comment by mbriskar** [comment]
Is test failing or passing here?

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/246/
Test PASSed.

**Comment by mbriskar** [comment]
I think the test non deterministically fails

**Comment by mbriskar** [comment]
It was, it had the same phase as AST. Now it should work

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/877/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-base-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/877/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-base-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/877/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport/org.jboss.windup.rules.apps.condition/FileContentTest/testFileContentScan/'><strong>org.jboss.windup.rules.apps.condition.FileContentTest.testFileContentScan</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/247/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/878/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/248/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/879/
Test PASSed.

File: rules-base/tests/src/test/resources/filecontentstest/WriteToFile.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import java.io.PrintStream;
+import java.io.IOException;
+import java.io.File;
+
+import java.util.Random;
+
+public class WriteToFile
```
Comment (jsight) [inline rules-base/tests/src/test/resources/filecontentstest/WriteToFile.java]: What is this used for?

**Pull request closed without merging**